generator/lib/src/realm_field_info.dart

```diff
@@ -27,15 +27,15 @@ class RealmFieldInfo {
   final FieldElement fieldElement;
   final String? mapTo;
   final bool isPrimaryKey;
-  final bool indexed;
+  final RealmIndexType indexType;
```

Comment:
we could preserve the full `Indexed` annotation here
```suggestion
  final Indexed? index;
```

Reply:
I don't think we should. We don't preserve the annotations for other things, such as `mapTo` or `isPrimaryKey`, so I'd rather extract the meaningful information.

Reply:
the annotation `Indexed` is just a data bag class in here

Reply:
instead of having a nullable enum we could just have an annotation instance here

Reply:
Right, but the only meaningful thing it contains is the `indexType`. You could make the same argument for the `MapTo` annotation - it is similarly a data bag with a single field. In any case, I would prefer that we extract the annotation data since it matches better the expectations I have of `RealmFieldInfo` - it's just a container for all the information that describes a field in Realm. The fact certain information is obtained through annotations is kind of an implementation detail and not something we need here.

Reply:
Ok lets keep it